### PR TITLE
Replace StayAliveObjective with EscapeShuttleObjective.

### PR DESCRIPTION
**Pull request by Vordenburg**
## About the PR 

This PR adds in the classic Escape on the Emergency Shuttle objective to supercede the Stay Alive objective.

In order to succeed at this objective, you must:

- be on the emergency shuttle,
- be alive, and
- not be cuffed.

**Changelog**

:cl:
- add: Added the Escape on the Emergency Shuttle objective.
- remove: Removed the Stay Alive objective.

### CODE REVIEW

File: Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                var agentIsAlive = !_mind.CharacterDeadIC;
+                var agentIsEscaping = true;
+
+                if (entMan.TryGetComponent<CuffableComponent>(_mind.OwnedEntity, out var cuffed)
+                    && cuffed.CuffedHandCount > 0)
+                    // You're not escaping if you're restrained!
+                    agentIsEscaping = false;
+
+                if (_stationData == null) {
+                    // Fallback and let the agent escape on any emergency shuttle.
```
Comment (Elijahrane) [inline Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs:74]: For multistation support shouldn't this always be how it works?

**Comment by Zumorica** [comment]
Why delete the StayAlive objective entirely? I think both can and should coexist (even if they're mutually exclusive at the time of picking objectives)

File: Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs (hunk 1 above)
Comment (Vordenburg) [reply Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs:74]: That could be possible. I haven't played any rounds that had multiple player stations, so I can't say. If that's the better way to handle this, then it'd be simpler than saving what station the entity spawns in.

**Comment by mirrorcult** [comment]
'staying alive' is pretty much just an unequivocally less interesting version of escaping on the shuttle, imo. i think its fine to remove it in favor of this

**Comment by Zumorica** [comment]
I think 'staying alive' offers much more flexibility and interesting player choices as opposed to just having 'escape on the shuttle':
Perhaps you'll stay on the station, or board the emergency shuttle. Maybe when things inevitably go wrong in the emergency shuttle due to a failed assassination attempt you can space yourself and survive *just enough* to see the round end and get that sweet sweet objective completion. This objective is only as boring as the player wants to make it, in my opinion

**Comment by 0x6273** [comment]
I think escaping on a different shuttle should count as well.

**Comment by keronshb** [comment]
Escape on the shuttle objective can be a bit dicey as it can encourage either:
A) Podbabies
B) Shuttle Murder City/Yakity Sax
C) Doing 100% stealth the entire round

**Comment by theashtronaut** [comment]
The "Escape on the shuttle" objective is less awful when you consider /tg/ has escape pods as well which satisfy the requirement. It means if your 'cover' is fully blown you can still escape without having to slaughter the entire crew if they catch wind that you're on the escape shuttle with them.

That kind of thing (escape pods) could always be looked at in the future if desired. I generally favor the escape objective over just "Stay alive" because to stay alive you can just hide in a locker for the rest of the round after completing your other objectives and "win" without any further interaction in the game.

File: Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs (hunk 1 above)
Comment (moonheart08) [reply Content.Server/Objectives/Conditions/EscapeShuttleCondition.cs:74]: I'd prefer this always allows any escape shuttle, as I'd like a true multistation map or two later.

**Review by mirrorcult — CHANGES_REQUESTED** [review]
check for any shuttle as mentioned above

**Comment by 0x6273** [comment]
So is there a reason why escaping on a different shuttle shouldn't count? You're reaching centcom either way so why does it matter how you got there?

**Comment by mirrorcult** [comment]
> So is there a reason why escaping on a different shuttle shouldn't count? You're reaching centcom either way so why does it matter how you got there?

image

**Review by mirrorcult — APPROVED** [review]

**Comment by 0x6273** [comment]
> > So is there a reason why escaping on a different shuttle shouldn't count? You're reaching centcom either way so why does it matter how you got there?
> 
> image

The code is still using `stationData.EmergencyShuttle`. If you FTL on another shuttle it won't count.

**Comment by mirrorcult** [comment]
it iterates over every station

**Comment by mirrorcult** [comment]
oh, you mean the cargo shuttle

**Comment by 0x6273** [comment]
> oh, you mean the cargo shuttle

But not every shuttle, only emergency shuttles.

**Comment by mirrorcult** [comment]
a better check would probably be just that they're on the centcomm map (in addition to the alive/cuff) check. that way it works with pods as well

**Comment by 0x6273** [comment]
> oh, you mean the cargo shuttle

Yes, or a player-built shuttle.

**Comment by LordEclipse** [comment]
> a better check would probably be just that they're on the centcomm map (in addition to the alive/cuff) check. that way it works with pods as well

The round ends as soon as the evac shuttle reaches centcomm, so that check would be a little odd. Maybe a check that they are in transit to centcomm instead?

**Pull request merged**